src/Omega_h_adios2.cpp

```diff
@@ -70,14 +84,12 @@ static void write_array(adios2::IO &io, adios2::Engine &writer, Mesh* mesh,
   for (size_t i = 0; i<array.size(); ++i)
      myData.push_back(array.data()[i]);
 
-  const std::size_t Nx = myData.size();
-  std::string subname = name + "_size";
-  write_value(io, writer, mesh->comm(), Nx, subname);
+  const std::size_t Nx = myData.size()/ncomp;
+  long unsigned int n_comp = static_cast<long unsigned int>(ncomp);
 
-  subname = name + "_data";
   adios2::Variable<T> bpData = 
 	  io.DefineVariable<T>(
-          subname, {comm_size * Nx}, {rank * Nx}, {Nx}, adios2::ConstantDims);
+          name, {comm_size * Nx, n_comp}, {rank * Nx,0}, {Nx, n_comp}, adios2::ConstantDims);
```

Comment:
I'm pretty sure this will break in parallel since `Nx` is different on every rank.